src/binaryutils.cpp

```diff
@@ -151,18 +151,18 @@ Status validateNumberOfChannels(const std::shared_ptr<TensorInfo>& tensorInfo,
     }
 
     // At this point we can either have nhwc format or pretendant to be nhwc but with ANY layout in pipeline info
-    size_t numberOfChannels = 0;
-    if (tensorInfo->getEffectiveShape().size() == 4) {
-        numberOfChannels = tensorInfo->getEffectiveShape()[3];
-    } else if (tensorInfo->isInfluencedByDemultiplexer() && tensorInfo->getEffectiveShape().size() == 5) {
-        numberOfChannels = tensorInfo->getEffectiveShape()[4];
+    dimension_value_t numberOfChannels = DYNAMIC_DIMENSION;
+    if (tensorInfo->getShape_3().size() == 4) {
+        numberOfChannels = tensorInfo->getShape_3()[3].getAnyValue();
```

Comment:
`getAnyValue` will be removed and should not be used here. If you desire to use `getAnyValue` you always need to consider two options - color dimension is dynamic or static. That's why we have Dimension::isDynamic/Dimension::isStatic methods.

```
bool isColorStatic = tensorInfo->getShape_3()[3].isStatic()
```

In this case, the number of channels can be static or dynamic and you only implemented the static path. What if model accepts (1, 224, 224, [1~3]) shape? We should validate the number of channels (`input.channels()`) against color dimension range.

```
if (tensorInfo->getShape_3()[3].isStatic()) {
  if (tensorInfo->getShape_3()[3].getStaticValue() != input.channels()) ...
} else if (tensor->getShape_3()[3] == Dimension::any()) {
  // any will do. do nothing
} else {
  if (tensorInfo->getShape_3()[3].getMinValue() > input.channels() || tensorInfo->getShape_3()[3].getMaxValue() < input.channels()) ...
}
```

Reply:
Support binary inputs with dynamic models should be done as separate ticket. For now check and if dynamic then return !status.ok(). We need to implement logic here what should be the final dimension.

Reply:
Task added